composer/utils/object_store/oci_object_store_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Utilities for monkey patching OCI."""
+
+import threading
+
+from composer.utils.import_helpers import MissingConditionalImportError
+
+
+def should_retry(self, exception=None, response=None, **kwargs):
+    """Uses OCI retry and adds BaseRequestException.
+
+    https://github.com/oracle/oci-python-sdk/blob/8ba39d63242fd406fb7a97fb6f9acdde81eb6dd9/src/oci/retry/retry_checkers.py#L166
+    """
+    try:
+        from circuitbreaker import CircuitBreakerError
+        from oci._vendor.requests.exceptions import ConnectionError as RequestsConnectionError
+        from oci._vendor.requests.exceptions import RequestException as BaseRequestException
+        from oci._vendor.requests.exceptions import Timeout
+        from oci.exceptions import ConnectTimeout, RequestException, ServiceError
+    except ImportError as e:
+        raise MissingConditionalImportError(
+            conda_package='oci',
+            extra_deps_group='oci',
+            conda_channel='conda-forge',
+        ) from e
+    print(type(exception))
+    print(str(exception))
+    if isinstance(exception, Timeout):
```

Comment:
You can also just group these into a `isinstance(exception, (Timeout....)` for all the valid exception types

Reply:
This is a copy paste from OCI's code so would prefer to keep it as close to the original as possible to get cleaner diffs